app/models/aspace_version_control/get_agents_job.rb

```diff
@@ -0,0 +1,235 @@
+      process_all_cpf_files('people', output_dir, chunk_size: chunk_size, start_from: start_from)
+    end
+
+    def process_all_family_cpf_files(output_dir, chunk_size: 500, start_from: 0)
+      process_all_cpf_files('families', output_dir, chunk_size: chunk_size, start_from: start_from)
+    end
+
+    def process_all_corporate_entity_cpf_files(output_dir, chunk_size: 500, start_from: 0)
+      process_all_cpf_files('corporate_entities', output_dir, chunk_size: chunk_size, start_from: start_from)
+    end
+
+    private
+
+    def get_agent_ids_by_type(agent_type)
+      case agent_type
+      when 'people'
+        list_person_agents
+      when 'families'
+        list_family_agents
+      when 'corporate_entities'
+        list_corporate_entities_agents
+      else
+        raise "Unknown agent type: #{agent_type}"
+      end
+    end
+
+    def resolve_start_position(checkpoint_file, start_from)
+      if File.exist?(checkpoint_file) && start_from == 0
+        checkpoint_position = File.read(checkpoint_file).to_i
+        Rails.logger.info("Resuming from checkpoint: #{checkpoint_position}")
+        checkpoint_position
+      else
+        start_from
+      end
+    end
+
+    def process_agent_batch(agent_ids, agent_type, output_dir, start_from, total_count, chunk_size, checkpoint_file)
+      processed_count = 0
+
+      agent_ids.drop(start_from).each_with_index do |id, relative_index|
+        absolute_index = start_from + relative_index
+
+        processed_count += 1 if process_single_agent(id, agent_type, output_dir, absolute_index)
+
+        handle_progress_and_checkpoints(absolute_index, total_count, agent_type, chunk_size, checkpoint_file)
+      end
+
+      processed_count
+    end
+
+    def process_single_agent(id, agent_type, output_dir, absolute_index)
+      write_cpf_to_file(output_dir, id, agent_type)
+      true
+    rescue => error
+      Rails.logger.error("Failed to process #{agent_type} agent #{id} at index #{absolute_index}: #{error.message}")
+      @errors << "#{agent_type} agent #{id}: #{error.message}"
+      false
+    end
+
+    def handle_progress_and_checkpoints(absolute_index, total_count, agent_type, chunk_size, checkpoint_file)
+      # Progress logging every 100 records
+      Rails.logger.info("Processed #{absolute_index + 1}/#{total_count} #{agent_type} agents") if (absolute_index + 1) % 100 == 0
+
+      # Save checkpoint every chunk_size records
+      if (absolute_index + 1) % chunk_size == 0
+        File.write(checkpoint_file, absolute_index + 1)
+        Rails.logger.info("Checkpoint saved at #{absolute_index + 1}")
+        sleep(2) # Longer pause every chunk to prevent API overload
+      elsif (absolute_index + 1) % 20 == 0
+        sleep(0.05)
+      end
+    end
+
+    def cleanup_checkpoint(checkpoint_file)
+      File.delete(checkpoint_file) if File.exist?(checkpoint_file)
+    end
+
+    def generate_cpf_filename(doc, id)
+      namespace = { 'eac' => 'urn:isbn:1-931666-33-4' }
+
```

Comment:
per @regineheberlein . concat name from name parts + concet with the aspace uri. for example
surname_forename_agent_type(e.g people, families, corporate_enitities)_id